src/Metadata/StorageInterface.php

```diff
@@ -0,0 +1,88 @@
+<?php
+
+namespace Tuf\Metadata;
+
+/**
+ * Defines an interface for saving and loading trusted TUF metadata.
+ */
+interface StorageInterface
+{
+    /**
+     * Loads trusted root metadata.
+     *
+     * @return \Tuf\Metadata\RootMetadata
+     *   The trusted root metadata.
+     *
+     * @throws \LogicException
+     *   Thrown if the root metadata cannot be loaded.
+     */
+    public function getRoot(): RootMetadata;
+
+    /**
+     * Persists the trusted root metadata.
+     *
+     * @param \Tuf\Metadata\RootMetadata $metadata
+     *   The root metadata to persist. Must be trusted.
+     */
+    public function saveRoot(RootMetadata $metadata): void;
```

Comment:
There 4 `save*` methods . `saveTargets` has a different comment for the arg but otherwise there are the same.

Did you think about having just 

`public function saveMetaddata(MetadataBase $metadata): void;`

and then the implementations calling `private/protected` methods based on type?

Reply:
I think that's a great idea. Implemented, although I didn't bother with specific protected methods based on type, because that felt like overkill at this stage. We can always refactor that in later if we need it.